lib/Controller/DisplayGroup.php

```diff
@@ -1736,4 +1749,181 @@ public function command($displayGroupId)
+     *      name="displayGroupId",
+     *      in="path",
+     *      description="The Display Group ID",
+     *      type="integer",
+     *      required=true
+     *   ),
+     *  @SWG\Parameter(
+     *      name="displayGroup",
+     *      in="formData",
+     *      description="The name for the copy",
+     *      type="string",
+     *      required=true
+     *   ),
+     *  @SWG\Parameter(
+     *      name="description",
+     *      in="formData",
+     *      description="The description for the copy",
+     *      type="string",
+     *      required=false
+     *   ),
+     *  @SWG\Parameter(
+     *      name="copyMembers",
+     *      in="formData",
+     *      description="Flag indicating whether to copy all display and display group members",
+     *      type="integer",
+     *      required=false
+     *   ),
+     *  @SWG\Parameter(
+     *      name="copyAssignments",
+     *      in="formData",
+     *      description="Flag indicating whether to copy all layout and media assignments",
+     *      type="integer",
+     *      required=false
+     *   ),
+     *  @SWG\Parameter(
+     *      name="copyTags",
+     *      in="formData",
+     *      description="Flag indicating whether to copy all tags",
+     *      type="integer",
+     *      required=false
+     *   ),
+     *  @SWG\Response(
+     *      response=201,
+     *      description="successful operation",
+     *      @SWG\Schema(ref="#/definitions/DisplayGroup"),
+     *      @SWG\Header(
+     *          header="Location",
+     *          description="Location of the new record",
+     *          type="string"
+     *      )
+     *  )
+     * )
+     */
+    public function copy($displayGroupId)
+    {
+        // get display group object
+        /** @var \Xibo\Entity\DisplayGroup $displayGroup */
+        $displayGroup = $this->displayGroupFactory->getById($displayGroupId);
+
+        // get an array of assigned displays
+        /** @var Display[] $membersDisplays */
+        $membersDisplays = $this->displayFactory->getByDisplayGroupId($displayGroupId);
+
+        // get an array of assigned display groups
+        /** @var \Xibo\Entity\DisplayGroup[] $membersDisplayGroups */
+        $membersDisplayGroups = $this->displayGroupFactory->getByParentId($displayGroupId);
+
+        // get an array of assigned layouts
+        /** @var \Xibo\Entity\Layout[] $assignedLayouts */
+        $assignedLayouts = $this->layoutFactory->getByDisplayGroupId($displayGroupId);
+
+        // get an array of assigned media files
+        /** @var Media[] $assignedFiles */
+        $assignedFiles = $this->mediaFactory->getByDisplayGroupId($displayGroupId);
+
+        $copyMembers = $this->getSanitizer()->getCheckbox('copyMembers', 0);
+        $copyTags = $this->getSanitizer()->getCheckbox('copyTags', 0);
+        $copyAssignments = $this->getSanitizer()->getCheckbox('copyAssignments', 0);
+
+        if (!$this->getUser()->checkEditable($displayGroup)) {
```

Comment:
We could make this more efficient if we did the permissions check before we queried for the rest of the data